### ISSUE TITLE
wrongly cleanup previous supervisord's pid file

### ISSUE THREAD

**Issue by tyronecai** [Supervisor/supervisor#613]
Rerun supervisord cmd will wrongly cleanup previous supervisord pidfile.

```
def cleanup(self):
    for config, server in self.httpservers:
        if config['family'] == socket.AF_UNIX:
            if self.unlink_socketfiles:
                socketname = config['file']
                self._try_unlink(socketname)
    self._try_unlink(self.pidfile)
```

function self._try_unlink should check self.write_pidfile has been done

**Comment by mnaberez** [comment]
I don't see anything wrong with this code.  Why do you think this code is wrong?  Did it cause a problem for you?

**Comment by tyronecai** [comment]
Sorry for my previous description.
When the "supervisord" daemon is running, I wrongly repeat the command, then the  current pidfile will be cleared.
So when function "self._try_unlink" is called, it should check whether "self.write_pidfile" has been done.

**Comment by mnaberez** [comment]
Fixed in #879 (eadf59b6367d37318ec6168304e6fd3873491410).
